dace/codegen/targets/rtl.py

```diff
@@ -358,63 +565,139 @@ def unparse_tasklet(self, sdfg: sdfg.SDFG, dfg: state.StateSubgraphView,
-                    if self.verilator_debug else ""))
-            self.cpp_general_header_added = True
-        sdfg.append_global_code(
-            cpp_code=RTLCodeGen.CPP_MODEL_HEADER_TEMPLATE.format(
-                name=unique_name))
-
-        # add main cpp code to stream
-        callsite_stream.write(contents=RTLCodeGen.CPP_MAIN_TEMPLATE.format(
-            name=unique_name,
-            inputs=inputs,
-            outputs=outputs,
-            num_elements=num_elements,
-            vector_init=vector_init,
-            internal_state_str=internal_state_str,
-            internal_state_var=internal_state_var,
-            debug_sim_start="std::cout << \"SIM {name} START\" << std::endl;"
-            if self.verilator_debug else "",
-            debug_feed_element="std::cout << \"feed new element\" << std::endl;"
-            if self.verilator_debug else "",
-            debug_export_element="std::cout << \"export element\" << std::endl;"
-            if self.verilator_debug else "",
-            debug_internal_state="""
-// report internal state 
-VL_PRINTF("[t=%lu] clk_i=%u rst_i=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->clk_i, model->rst_i, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
+                cpp_code=RTLCodeGen.CPP_MODEL_HEADER_TEMPLATE.format(
+                    name=unique_name))
+
+            # add main cpp code to stream
+            callsite_stream.write(contents=RTLCodeGen.CPP_MAIN_TEMPLATE.format(
+                name=unique_name,
+                inputs=inputs,
+                outputs=outputs,
+                num_elements=str.join('\n', num_elements),
+                vector_init=vector_init,
+                valid_zeros=str.join('\n', valid_zeros),
+                ready_zeros=str.join('\n', ready_zeros),
+                read_input_hs=str.join('\n', read_input_hs),
+                feed_elements=str.join('\n', feed_elements),
+                in_ptrs=str.join('\n', in_ptrs),
+                out_ptrs=str.join('\n', out_ptrs),
+                export_elements=str.join('\n', export_elements),
+                write_output_hs=str.join('\n', write_output_hs),
+                hs_flags=str.join('\n', hs_flags),
+                input_hs_toggle=str.join('\n', input_hs_toggle),
+                output_hs_toggle=str.join('\n', output_hs_toggle),
+                running_condition=str.join(' && ', running_condition),
+                internal_state_str=internal_state_str,
+                internal_state_var=internal_state_var,
+                debug_sim_start="std::cout << \"SIM {name} START\" << std::endl;"
+                if self.verilator_debug else "",
+                debug_internal_state="""
+// report internal state
+VL_PRINTF("[t=%lu] ap_aclk=%u ap_areset=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->ap_aclk, model->ap_areset, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
```

Comment:
yapf

Reply:
Fixed in https://github.com/carljohnsen/dace/commit/37a8119b8b02530a60dbdbe4936d3735a4f113dd